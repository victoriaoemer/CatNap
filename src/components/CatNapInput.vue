<template>
  <div class="gradient-border-bottom w-full">
    <input
      class="bg-transparent focus:outline-none pb-1"
      :type="type"
      :placeholder="placeholder"
      v-model="inputValue"
    />
  </div>
</template>

<script setup lang="ts">
import { computed } from 'vue'

const props = defineProps({
  modelValue: {
    type: String,
    required: true,
  },
  type: {
    type: String,
    default: 'text',
  },
  placeholder: {
    type: String,
    default: '',
  },
})

const emit = defineEmits(['update:modelValue'])

const inputValue = computed({
  get() {
    return props.modelValue
  },
  set(value) {
    emit('update:modelValue', value)
  },
})
</script>
