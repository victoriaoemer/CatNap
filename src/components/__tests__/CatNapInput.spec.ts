import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import CatNapInput from '@/components/CatNapInput.vue';

describe('CatNapInput.vue', () => {
  it('renders the correct placeholder and type', () => {
    const wrapper = mount(CatNapInput, {
      props: {
        modelValue: '',
        placeholder: 'Enter text...',
        type: 'password',
      },
    });

    const input = wrapper.find('input');
    expect(input.attributes('placeholder')).toBe('Enter text...');
    expect(input.attributes('type')).toBe('password');
  });

  it('binds v-model correctly', async () => {
    const wrapper = mount(CatNapInput, {
      props: {
        modelValue: '',
      },
    });

    const input = wrapper.find('input');
    await input.setValue('test value');

    expect(wrapper.emitted()['update:modelValue'][0]).toEqual(['test value']);
  });

  it('applies the correct classes when settings is true', () => {
    const wrapper = mount(CatNapInput, {
      props: {
        modelValue: '',
        settings: true,
      },
    });

    expect(wrapper.find('div').classes()).toContain('bg-[#3B32AB]');
    expect(wrapper.find('div').classes()).toContain('border-[#3E2FE9]');
  });

  it('applies the correct classes when settings is false', () => {
    const wrapper = mount(CatNapInput, {
      props: {
        modelValue: '',
        settings: false,
      },
    });

    expect(wrapper.find('div').classes()).toContain('gradient-border-bottom');
  });

  it('matches the snapshot', () => {
    const wrapper = mount(CatNapInput, {
      props: {
        modelValue: '',
        placeholder: 'Enter text...',
        type: 'password',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
