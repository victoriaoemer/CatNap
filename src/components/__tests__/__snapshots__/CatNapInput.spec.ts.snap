// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CatNapInput.vue > matches the snapshot 1`] = `"<div class="gradient-border-bottom w-full"><input class="bg-transparent focus:outline-none pb-1" type="password" placeholder="Enter text..."></div>"`;
